ratis-logservice/src/main/java/org/apache/ratis/logservice/api/RecordListener.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.ratis.logservice.api;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Interface that, when registered with a {@link LogStream}, will receive all records written
+ * to that LogStream until it is removed.
+ */
+public interface RecordListener {
```

Comment:
Will more methods to be added? If not, is a simple `Consumer` sufficient as listeners (consumers)?

Reply:
Good point. Given what we know now, I don't think we have more complex requirements. Switching this over to a consumer is a good idea, I think.